libandroid-navigation/src/main/java/com/mapbox/services/android/navigation/v5/navigation/NavigationTelemetry.java

```diff
@@ -124,11 +116,13 @@ public void onArrival(RouteProgress routeProgress) {
   void initialize(@NonNull Context context, @NonNull String accessToken,
                   MapboxNavigation navigation, LocationEngine locationEngine) {
     if (!isInitialized) {
-      updateLocationEngineName(locationEngine);
-
       validateAccessToken(accessToken);
+      updateLocationEngineName(locationEngine);
+      DepartEventHandler departEventHandler = new DepartEventHandler(context);
+      departEventFactory = new DepartEventFactory(departEventHandler);
       this.context = context;
       NavigationMetricsWrapper.init(context, accessToken, BuildConfig.MAPBOX_NAVIGATION_EVENTS_USER_AGENT);
+      NavigationMetricsWrapper.toggleLogging(true);
```

Comment:
Just a reminder that we need to remove this after testing that this is working as expected and before merging.